modules/carousel/jetpack-carousel.js

```diff
@@ -435,6 +435,42 @@ jQuery(document).ready(function($) {
 		}
 	};
 
+	var processSingleImageGallery = function() {
+		// process links that contain img tag with attribute data-attachment-id
+		$( 'a img[data-attachment-id]' ).each(function() {
+			var container = $( this ).parent();
+
+			// skip if image was already added to gallery by shortcode
+			if( container.parent( '.gallery-icon' ).length ) {
+				return;
+			}
+
+			var valid = false;
+
+			// if link points to 'Media File' (ignoring GET parameters) and flag is set allow it
+			if ( $( container ).attr( 'href' ).split( '?' )[0] === $( this ).attr( 'data-orig-file' ).split( '?' )[0] &&
```

Comment:
@thingalon I have ported your changes from #6296 into this PR, can you please take a look?